src/Rox/Core/Core.php

```diff
@@ -221,11 +221,12 @@ public function setup(
     {
         $roxyUrl = ($roxOptions != null) ? $roxOptions->getRoxyURL() : null;
         if ($roxyUrl == null) {
-            $validApiKeyPattern = "/^[a-f\\d]{24}$/i";
+            $mongoApiKeyPattern = "/^[a-f\\d]{24}$/i";
```

Comment:
I don't know how to write PHP, so this is mostly just guesswork. I couldn't figure out how to write a test. There is currently a test to check for an error if there is an invalid API key, but no tests for no-error-if-the-key-is-vaild.